src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```
@@ -0,0 +1,197 @@
+            throw new CommandException(MESSAGE_NOT_FOUND);
+        }
+
+        else if (areaToSearch == areaIsName) {
+            String nameToSearch = searchPatientDescriptor.getName().toString().trim().toLowerCase();
+            Person personFound = personList.get(0);
+            boolean isPersonFound = false;
+
+            for (Person person : personList) {
+                String personName = person.getName().toString().trim().toLowerCase();
+                if (personName.equals(nameToSearch)) {
+                    personFound = person;
+                    isPersonFound = true;
+
+                }
+            }
+            if (!isPersonFound) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(String.format(MESSAGE_SEARCH_PERSON_SUCCESS, personFound));
+            }
+        }
+
+        else if (areaToSearch == areaIsTemperature) {
+            double startingTemperature = searchPatientDescriptor.getTemperatureRange().getStartingTemperature();
+            double endingTemperature = searchPatientDescriptor.getTemperatureRange().getEndingTemperature();
+            for (Person person : personList) {
+                if (person.getTemperature().getValue() >= startingTemperature
+                        && person.getTemperature().getValue() <= endingTemperature) {
+                    personWithinTemperatureRange.add(person);
+                }
+            }
+
+            if (personWithinTemperatureRange.isEmpty()) {
+                throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+            } else {
+                return new CommandResult(MESSAGE_SEARCH_PERSON_LIST_SUCCESS
+                        + getListOutput(personWithinTemperatureRange));
+            }
+        } else if (areaToSearch == tooManyArea) {
+            return new CommandResult(Messages.MESSAGE_TOO_MANY_COMMANDS);
+        }
+        return null;
```

I'm not too sure but I think the execution will not reach this return statement? If that is the case, I think we can throw a `CommandException` instead of returning `null`? Returning `null` here may cause a `NullPointerException` somewhere else.

---

Definitely true